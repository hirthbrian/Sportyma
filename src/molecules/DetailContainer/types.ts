import { ImageSourcePropType } from "react-native";

export interface Props {
  title: string;
  subTitle: string;
  icon: ImageSourcePropType;
  onPress?: (event: GestureResponderEvent) => void;
}
