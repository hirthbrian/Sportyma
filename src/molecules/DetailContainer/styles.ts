import styled from "styled-components/native";

import { Colors } from "../../types";

interface ImageProps {
  color: string;
}

export const Container = styled.Pressable`
  padding-top: 15px;
  flex-direction: row;
`;

export const Day = styled.Text`
  color: ${Colors.White};
  padding: 4px 10px;
`;

export const Image = styled.Image`
  width: 18px;
  height: 18px;
  tint-color: ${(props: ImageProps) => props.color};
`;
