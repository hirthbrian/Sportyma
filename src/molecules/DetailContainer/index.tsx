import { View } from "react-native";

import Circle from "../../atoms/Circle";
import Text from "../../atoms/TextKanit";

import { Container, Image } from "./styles";

import { Props } from "./types";

const DetailContainer = ({ title, subTitle, icon, onPress }: Props) => {
  return (
    <Container onPress={onPress}>
      <Circle size={36} color="#E3E8F9">
        <Image source={icon} color="#2559F2" />
      </Circle>
      <View>
        <Text>{title}</Text>
        <Text weight={300} color="#778EB9">
          {subTitle}
        </Text>
      </View>
    </Container>
  );
};

export default DetailContainer;
