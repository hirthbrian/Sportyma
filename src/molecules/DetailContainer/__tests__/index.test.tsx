import React from "react";
import renderer from "react-test-renderer";

import DetailContainer from "..";

const icon = require("../../../../assets/icons/calendar.png");

describe("<DetailContainer />", () => {
  let props;

  beforeEach(() => {
    props = {
      title: "test",
      subTitle: "sub test",
      icon,
    };
  });

  it("renders correcly", () => {
    expect(
      renderer.create(<DetailContainer {...props} />).toJSON()
    ).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer.create(<DetailContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
