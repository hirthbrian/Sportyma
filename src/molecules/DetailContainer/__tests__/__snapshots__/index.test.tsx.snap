// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailContainer /> match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingTop": 15,
      },
    ]
  }
>
  <View
    color="#E3E8F9"
    size={36}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#E3E8F9",
          "borderRadius": 18,
          "height": 36,
          "justifyContent": "center",
          "marginRight": 10,
          "width": 36,
        },
      ]
    }
  >
    <Image
      color="#2559F2"
      source={1}
      style={
        Array [
          Object {
            "height": 18,
            "tintColor": "#2559F2",
            "width": 18,
          },
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#152D4B",
            "fontFamily": "Kanit-Regular",
            "fontSize": 16,
          },
        ]
      }
    >
      test
    </Text>
    <Text
      color="#778EB9"
      style={
        Array [
          Object {
            "color": "#778EB9",
            "fontFamily": "Kanit-Light",
            "fontSize": 16,
          },
        ]
      }
      weight={300}
    >
      sub test
    </Text>
  </View>
</View>
`;
